scipy/linalg/decomp_svd.py

```diff
@@ -279,6 +280,97 @@ def diagsvd(s, M, N):
         raise ValueError("Length of s must be M or N.")
 
 
+def hosvd(a, full_tensor=True, check_finite=True):
```

Comment:
Not sure here (about which one) but consider having keyword only parameters.
```suggestion
def hosvd(a, *, full_tensor=True, check_finite=True):
```